### PR TITLE
Skip date field validation if empty

### PR DESCRIPTION
**Pull request by jclausen**
Empty date fields should not prevent submission (e.g. - when there is a publish up/publish down date setting for a custom object)

### CODE REVIEW

**Comment by blakecallens** [comment]
Thanks for pointing this out. I agree with the intent, but the proposed change could result in JS errors.

The parsableDate filter takes the readable date from the text input part of the date selector and turns it into an ISO 8601 datetime string that can be parsed by JS Date in all browsers. This is used by all date inputs in the admin section, including forms where a date is required.

The place to make the change for custom objects would be in the [angular controller for object_form](https://github.com/pencilblue/pencilblue/blob/master/plugins/pencilblue/templates/angular/admin/content/objects/object_form.html#L92).

**Comment by blakecallens** [comment]
Sorry, I totally read the commit backwards and thought you took out protection instead of added it. Merged.

File: public/js/angular/filters/parsable_date.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 angular.module('parseDate', [])
 .filter('parsableDate', function() {
     return function(input) {
+        if(typeof(input)==='undefined')
```
Comment (blakecallens) [inline public/js/angular/filters/parsable_date.js]: For future reference, we wrap all conditionals in brackets, I'll make the change in master.

**Comment by jclausen** [comment]
No worries. Thanks for the fast turnaround.  I'll make sure to wrap with brackets for any future pull requests.  -J

**Pull request merged**